stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -732,6 +809,71 @@ end
     return matches
 end
 
+# provide completion for keyword arguments in function calls
+function complete_keyword_argument(partial, last_idx, context_module)
+    fail = Completion[], 0:-1
+
+    # Quickly abandon if the situation does not look like the completion of a kwarg
+    idx_last_punct = something(findprev(x -> ispunct(x) && x != '_' && x != '!', partial, last_idx), 0)::Int
+    idx_last_punct == 0 && return fail
+    last_punct = partial[idx_last_punct]
+    last_punct == ',' || last_punct == ';' || last_punct == '(' || return fail
+    before_last_word_start = something(findprev(in(non_identifier_chars), partial, last_idx), 0)
+    before_last_word_start == 0 && return fail
+    all(isspace, partial[nextind(partial, idx_last_punct):before_last_word_start]) || return fail
+    frange, method_name_end = find_start_brace(partial[1:idx_last_punct])
+    method_name_end ∈ frange || return fail
+
+    # At this point, we are guaranteed to be in a set of parentheses, possibly a function
+    # call, and the last word (currently being completed) has no internal dot (i.e. of the
+    # form "foo.bar") and is directly preceded by `last_punct` (one of ',', ';' or '(').
+    # Now, check that we are indeed in a function call
+    frange = first(frange):(last_punct==';' ? prevind(partial, idx_last_punct) : idx_last_punct)
+    s = replace(partial[frange], r"\!+([^=\(]+)" => s"\1") # strip preceding ! operator
+    ex = Meta.parse(s * ')', raise=false, depwarn=false)
+    isa(ex, Expr) || return fail
+    ex.head === :call || (ex.head === :. && ex.args[2] isa Expr && (ex.args[2]::Expr).head === :tuple) || return fail
+
+    # inlined `complete_methods` function since we need the `kwargs_ex` variable
+    func, found = get_value(ex.args[1], context_module)
+    !(found::Bool) && return fail
+    args_ex, kwargs_ex, kwargs_flag = complete_methods_args(ex.args[2:end], ex, context_module, true, true)
+
+    kwargs_flag == 2 && return fail # one of the previous kwargs is invalid
+
+    methods = Completion[]
+    complete_methods!(methods, Core.Typeof(func), args_ex, kwargs_ex, -1, kwargs_flag == 1)
+
+    # Finally, for each method corresponding to the function call, provide completions
+    # suggestions for each keyword that starts like the last word and that is not already
+    # used previously in the expression. The corresponding suggestion is "kwname="
+    # If the keyword corresponds to an existing name, also include "kwname" as a suggestion
+    # since the syntax `foo(; bar)` is equivalent to `foo(; bar=bar)`
+    wordrange = nextind(partial, before_last_word_start):last_idx
+    last_word = partial[wordrange] # the word to complete
+    kwargs = Set{String}()
+    for m in methods
+        m::MethodCompletion
+        possible_kwargs = Base.kwarg_decl(m.method)
```

Comment:
This can apparently crash in some cases (only builtins maybe?). For example:
```julia
julia> isdefined(res┌ Error: Error in the keymap
│   exception =
│    TypeError: in typeassert, expected Core.MethodTable, got a value of type Nothing
│    Stacktrace:
│      [1] get_methodtable
│        @ ./reflection.jl:1657 [inlined]
│      [2] kwarg_decl(m::Method, kwtype::Nothing) (repeats 2 times)
│        @ Base ./methodshow.jl:82
│      [3] complete_keyword_argument(partial::String, last_idx::Int64, context_module::Module)
│        @ REPL.REPLCompletions ~/julia/usr/share/julia/stdlib/v1.9/REPL/src/REPLCompletions.jl:861
...
```